__tests__/selectoptions.js

```diff
@@ -0,0 +1,190 @@
+      onMouseMove: eventsHandler,
+      onMouseDown: eventsHandler,
+      onFocus: eventsHandler,
+      onMouseUp: eventsHandler,
+      onClick: eventsHandler
+    };
+
+    const { getByTestId } = render(
+      <select data-testid="element">
+        <option {...eventHandlers} value="1">
+          1
+        </option>
+        <option {...eventHandlers} value="2">
+          2
+        </option>
+        <option {...eventHandlers} value="3">
+          3
+        </option>
+      </select>
+    );
+
+    userEvent.selectOptions(
+      getByTestId("element"),
+      option => option.value === "2"
+    );
+
+    expect(events[1]).toBe(undefined);
+    expect(events[3]).toBe(undefined);
+    expect(events[2]).toEqual([
+      "mouseover",
+      "mousemove",
+      "mousedown",
+      "focus",
+      "mouseup",
+      "click"
+    ]);
+  });
+
+  it("should fire the correct events on selected OPTION children with <select multiple>", () => {
+    function handleEvent(evt) {
+      const optValue = parseInt(evt.target.value);
+      events[optValue] = [...(events[optValue] || []), evt.type];
+    }
+
+    const events = [];
+    const eventsHandler = jest.fn(handleEvent);
+    const eventHandlers = {
+      onMouseOver: eventsHandler,
+      onMouseMove: eventsHandler,
+      onMouseDown: eventsHandler,
+      onFocus: eventsHandler,
+      onMouseUp: eventsHandler,
+      onClick: eventsHandler
+    };
+
+    const { getByTestId } = render(
+      <select multiple data-testid="element">
+        <option {...eventHandlers} value="1">
+          1
+        </option>
+        <option {...eventHandlers} value="2">
+          2
+        </option>
+        <option {...eventHandlers} value="3">
+          3
+        </option>
+      </select>
+    );
+
+    userEvent.selectOptions(
+      getByTestId("element"),
+      option => option.value === "1" || option.value === "3"
+    );
+
+    expect(events[2]).toBe(undefined);
+    expect(events[1]).toEqual([
+      "mouseover",
+      "mousemove",
+      "mousedown",
+      "focus",
+      "mouseup",
+      "click"
+    ]);
+
+    expect(events[3]).toEqual([
+      "mouseover",
+      "mousemove",
+      "mousedown",
+      "focus",
+      "mouseup",
+      "click"
+    ]);
+  });
+
+  it("sets the selected prop on the selected OPTION", () => {
+    const onSubmit = jest.fn();
+
+    const { getByTestId } = render(
+      <form onSubmit={onSubmit}>
+        <select multiple data-testid="element">
+          <option data-testid="val1" value="1">
+            1
+          </option>
+          <option data-testid="val2" value="2">
+            2
+          </option>
+          <option data-testid="val3" value="3">
+            3
+          </option>
+        </select>
+        <input type="submit" data-testid="submit" value="Submit" />
+      </form>
+    );
+
+    userEvent.selectOptions(
+      getByTestId("element"),
+      option => option.value === "1" || option.value === "3"
```

Comment:
I like the function approach, but I think it would be handy to be able to pass an array of values or a single value as well

`userEvent.selectOptions(element, ['1', '3'])`
`userEvent.selectOptions(element, '3')`

Reply:
Yeah, I think that's a good idea.  The function seemed best while I was trying different ideas, but I think a simple array or scalar value is more readable, and I don't foresee ever needing more than that.  And it also conforms more to the idea of writing tests from the perspective of a user.  

I can change that ASAP (most likely tonight).  Any thoughts as to whether I keep the ability to pass the function in there?  Can you think of any situation where you'd want to pass a function rather than just an array of values?

Reply:
Also will update the README

Reply:
I can't think of any but it shouldn't be a big deal to keep it

Reply:
I see now you've removed it. Let's keep it like that for now then